Explorer/Assets/DCL/Multiplayer/Movement/Systems/MultiplayerMovementMessageBus.cs

```diff
@@ -0,0 +1,184 @@
+        {
+            try
+            {
+                packet = packetParser.ParseFrom(data).EnsureNotNull();
+                return true;
+            }
+            catch (Exception e)
+            {
+                ReportHub.LogWarning(
+                    ReportCategory.ARCHIPELAGO_REQUEST,
+                    $"Someone sent invalid packet: {data.Length} {data.HexReadableString()} {e}"
+                );
+
+                packet = null;
+                return false;
+            }
+        }
+
+        private async UniTaskVoid HandleAsync(SmartWrap<Packet> packet, Participant participant)
+        {
+            using (packet)
+            {
+                await using ExecuteOnMainThreadScope _ = await ExecuteOnMainThreadScope.NewScopeAsync();
+
+                if (packet.value.Movement != null) // TODO (Vit): filter out Island messages if Participant is presented in the Room
+                {
+                    Decentraland.Kernel.Comms.Rfc4.Movement proto = packet.value.Movement;
+
+                    var message = new FullMovementMessage
+                    {
+                        timestamp = proto.Timestamp,
+                        position = new Vector3(proto.PositionX, proto.PositionY, proto.PositionZ),
+                        velocity = new Vector3(proto.VelocityX, proto.VelocityY, proto.VelocityZ),
+                        animState = new AnimationStates
+                        {
+                            MovementBlendValue = proto.MovementBlendValue,
+                            SlideBlendValue = proto.SlideBlendValue,
+                            IsGrounded = proto.IsGrounded,
+                            IsJumping = proto.IsJumping,
+                            IsLongJump = proto.IsLongJump,
+                            IsFalling = proto.IsFalling,
+                            IsLongFall = proto.IsLongFall,
+                        },
+                        isStunned = proto.IsStunned,
+                    };
+
+                    Inbox(message, participant.Identity);
+                }
+            }
+        }
+
+        private void Inbox(FullMovementMessage fullMovementMessage, string @for)
+        {
+            if (InboxByParticipantMap.TryGetValue(@for, out SimplePriorityQueue<FullMovementMessage>? queue) && !queue.Contains(fullMovementMessage))
+                queue.Enqueue(fullMovementMessage, fullMovementMessage.timestamp);
+            else
+            {
+                var newQueue = new SimplePriorityQueue<FullMovementMessage>(); // TODO (Vit): pooling
+                newQueue.Enqueue(fullMovementMessage, fullMovementMessage.timestamp);
+
+                InboxByParticipantMap.Add(@for, newQueue);
+            }
+        }
+
+        public void SelfSendWithDelay(FullMovementMessage message, float delay)
+        {
+            UniTask.Delay(TimeSpan.FromSeconds(delay))
+                   .ContinueWith(() => Inbox(message, @for: RemotePlayerMovementComponent.TEST_ID))
```

Comment:
Let's not create a delegate allocation, convert to the async `UniTaskVoid` method